django_project/certification/models.py

```diff
@@ -0,0 +1,302 @@
+            words = self.name.split()
+            filtered_words = [t for t in words if t.lower() not in STOP_WORDS]
+            # unidecode() represents special characters (unicode data) in ASCII
+            new_list = unidecode(' '.join(filtered_words))
+            self.slug = slugify(new_list)[:50]
+
+        super(TrainingCenter, self).save(*args, **kwargs)
+
+    def __unicode__(self):
+        return u'%s' % self.name
+
+    def get_absolute_url(self):
+        """Return URL to training center detail page
+        :return: URL
+        :rtype: str
+        """
+        return reverse('training-center-detail', kwargs={'slug': self.slug})
+
+
+class CourseType(models.Model):
+    """ Course Type model """
+    name = models.CharField(
+        help_text="Course type",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    slug = models.SlugField(unique=True)
+    objects = models.Manager()
+    training_center = models.ManyToManyField(
+        TrainingCenter,
+        through='Membership',
+        through_fields=('coursetype', 'trainingcenter')
+    )
+
+    # noinspection PyClassicStyleClass.
+    class Meta:
+        """ Meta class for Course types """
+        app_label = 'certification'
+        ordering = ['name']
+
+    def save(self, *args, **kwargs):
+        """ Overloaded save method
+        :param args:
+        :param kwargs:
+        """
+        if not self.pk:
+            words = self.name.split()
+            filtered_words = [t for t in words if t.lower() not in STOP_WORDS]
+            # unidecode() represents special characters (unicode data) in ASCII
+            new_list = unidecode(' '.join(filtered_words))
+            self.slug = slugify(new_list)[:50]
+
+        super(CourseType, self).save(*args, **kwargs)
+
+    def __unicode__(self):
+        return u'%s' % self.name
+
+
+class Membership(models.Model):
+    """ Many to many relationship for Course Type and Training Center """
+    coursetype = models.ForeignKey(CourseType, on_delete=models.CASCADE)
+    trainingcenter = models.ForeignKey(TrainingCenter, on_delete=models.CASCADE)
+
+
+class CourseConvener(models.Model):
+    """ Register Course Convener """
+    name = models.CharField(
+        help_text="First name of course conveners",
+        max_length=150,
+        null=False,
+        blank=False
+    )
+
+    surename = models.CharField(
+        help_text="Surename course conveners",
+        max_length=150,
+        null=False,
+        blank=False
+    )
+
+    email = models.CharField(
+        help_text="Email address",
+        max_length=200,
+        null=False,
+        blank=False
+    )
+
+    slug = models.SlugField(unique=True)
+    objects = models.Manager()
+
+    # noinspection PyClassicStyleClass.
+    class Meta:
+        """ Meta class for Course convener """
+        app_label = 'certification'
+        ordering = ['name']
+
+    def save(self, *args, **kwargs):
+        """ Overloaded save method
```

Comment:
Remove redundant spaces.
Full stop at end of docstring sentences.
New line after first line of docstring.